### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Oleh888**

### CODE REVIEW

File: src/main/java/core/basesyntax/Engine.java [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+        this.typeOfFuel = typeOfFuel;
+        this.power = power;
+        this.volume = volume;
+        this.price = price;
+        this.isNew = isNew;
+    }
+
+    public Engine clone(Engine other) throws CloneNotSupportedException {
+        return new Engine(other.typeOfFuel,
+                other.power, other.volume, other.price, other.isNew);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:20]: do we need other here?

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
 
+    public Engine(String typeOfFuel, int power, int volume, int price, boolean isNew) {
+        this.typeOfFuel = typeOfFuel;
+        this.power = power;
+        this.volume = volume;
+        this.price = price;
+        this.isNew = isNew;
+    }
+
+    public Engine clone(Engine other) throws CloneNotSupportedException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:18]: can we remove throws?

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
 
+    public Car(Engine engine, String color, int maxSpeed, int doors, boolean isNew) {
+        this.isNew = isNew;
+        this.doors = doors;
+        this.color = color;
+        this.maxSpeed = maxSpeed;
+        this.isNew = isNew;
+    }
+
+    public Car clone(Car other) throws CloneNotSupportedException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:18]: same

File: src/main/java/core/basesyntax/Engine.java (hunk 1 above)
Comment (Oleh888) [reply src/main/java/core/basesyntax/Engine.java]: corrected

File: src/main/java/core/basesyntax/Engine.java (hunk 2 above)
Comment (Oleh888) [reply src/main/java/core/basesyntax/Engine.java]: corrected

File: src/main/java/core/basesyntax/Car.java (hunk 3 above)
Comment (Oleh888) [reply src/main/java/core/basesyntax/Car.java]: corrected

File: src/main/java/core/basesyntax/Car.java [hunk 4]
```diff
@@ -1,5 +1,21 @@
...
+    public Car(Engine engine, String color, int maxSpeed, int doors, boolean isNew) {
+        this.isNew = isNew;
+        this.doors = doors;
+        this.color = color;
+        this.maxSpeed = maxSpeed;
+        this.isNew = isNew;
+    }
+
+    public Car clone() {
+        return new Car(engine, color, maxSpeed, doors, isNew);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:19]: you will get the new Car object, but what about it's engine?

Comment (Oleh888) [reply src/main/java/core/basesyntax/Car.java:19]: it's by mistake

Comment (Oleh888) [reply src/main/java/core/basesyntax/Car.java:19]: dome

File: src/main/java/core/basesyntax/Car.java [hunk 5]
```diff
@@ -1,5 +1,21 @@
...
+    public Car(Engine engine, String color, int maxSpeed, int doors, boolean isNew) {
+        this.engine = engine;
+        this.doors = doors;
+        this.color = color;
+        this.maxSpeed = maxSpeed;
+        this.isNew = isNew;
+    }
+
+    public Car clone() {
+        return new Car(engine, color, maxSpeed, doors, isNew);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:19]: you will get a new Car but with the same engine

Comment (Oleh888) [reply src/main/java/core/basesyntax/Car.java]: done

**Pull request closed without merging**